make/modules/java.base/gensrc/GensrcBuffer.gmk

```diff
@@ -28,363 +28,222 @@ ifeq ($(INCLUDE), true)
-$(eval $(call SetupGenBuffer,ByteBufferAsLongBufferB,   $(BYTE_X_BUF), type := long, BO := B))
-$(eval $(call SetupGenBuffer,ByteBufferAsLongBufferRB,  $(BYTE_X_BUF), type := long, RW := R, BO := B))
-$(eval $(call SetupGenBuffer,ByteBufferAsFloatBufferB,  $(BYTE_X_BUF), type := float, BO := B))
-$(eval $(call SetupGenBuffer,ByteBufferAsFloatBufferRB, $(BYTE_X_BUF), type := float, RW := R, BO := B))
-$(eval $(call SetupGenBuffer,ByteBufferAsDoubleBufferB, $(BYTE_X_BUF), type := double, BO := B))
-$(eval $(call SetupGenBuffer,ByteBufferAsDoubleBufferRB,$(BYTE_X_BUF), type := double, RW := R, BO := B))
+  $$(eval $$(call SetupStreamPreProcessing, GEN_BUFFER_$1, \
+      SOURCE_FILE := $$(BUFFER_INPUT_DIR)/$$($1_TEMPLATE).java.template, \
+      OUTPUT_FILE := $$($1_OUTPUT), \
+      INFO := Generating buffer class $1.java, \
+      KEYS := \
+          $$($1_TYPE) \
+          $$($1_RW_KEYS) \
+          bo$$($1_BYTE_ORDER) \
+          $$(call Conv, $$($1_TYPE), category) \
+          $$(call Conv, $$($1_TYPE), streams), \
+      REPLACEMENTS := \
+          type=$$($1_TYPE) \
+          BO=$$($1_BYTE_ORDER) \
+          RW=$$($1_RW_REPLACEMENT) \
+          LG_BYTES_PER_VALUE=$$($1_LBPV) \
+          BYTES_PER_VALUE="(1$$$$(SPACE)<<$$$$(SPACE)$$($1_$2_LBPV))" \
+          a=$$(call Conv, $$($1_TYPE), a) \
+          A=$$(call Conv, $$($1_TYPE), A) \
+          fulltype=$$(call Conv, $$($1_TYPE), fulltype) \
+          Fulltype=$$(call Conv, $$($1_TYPE), Fulltype) \
+          memtype=$$(call Conv, $$($1_TYPE), memtype) \
+          Memtype=$$(call Conv, $$($1_TYPE), Memtype) \
+          streamtype=$$(call Conv, $$($1_TYPE), streamtype) \
+          Streamtype=$$(call Conv, $$($1_TYPE), Streamtype) \
+          Type=$$(call Conv, $$($1_TYPE), Type) \
+          x=$$(call Conv, $$($1_TYPE), x) \
+          fromBits=$$(call Conv, $$($1_TYPE), fromBits, $$($1_BYTE_ORDER)) \
+          toBits=$$(call Conv, $$($1_TYPE), toBits, $$($1_BYTE_ORDER)) \
+          swap=$$(call Conv, $$($1_TYPE), swap, $$($1_BYTE_ORDER)) \
+          Swaptype=$$(call Conv, $$($1_TYPE), Swaptype, $$($1_BYTE_ORDER)), \
+  ))
+  TARGETS += $$(GEN_BUFFER_$1)
+  $1 += $$(GEN_BUFFER_$1)
+
+  ifeq ($$($1_GENERATE_BIN), true)
+    # Setup generation of snippet files, one for each non-byte type. This will
+    # populate $1_BIN_SNIPPET_FILES.
+    $1_BIN_SNIPPET_FILES :=
+    $$(foreach t, $$(NON_BYTE_NUMBER_TYPES), \
+      $$(eval $$(call SetupGenBufferBinSnippets,$1,$$t)) \
+    )
+
+    # Inject these snippets in the file generated by GEN_BUFFER_$1
+    $$($1_REAL_OUTPUT): $$($1_OUTPUT) $$($1_BIN_SNIPPET_FILES)
+	$$(call LogInfo, Concatenating buffer class bin snippets for $1)
+        # Delete everything from the line containing #BIN and below
+	$$(SED) -e '/#BIN/,$$$$d' < $$($1_OUTPUT) > $$($1_REAL_OUTPUT).tmp
+	$$(CAT) $$($1_BIN_SNIPPET_FILES) >> $$($1_REAL_OUTPUT).tmp
+	$$(ECHO) "}" >> $$($1_REAL_OUTPUT).tmp
+	$$(MV) $$($1_REAL_OUTPUT).tmp $$($1_REAL_OUTPUT)
```

Comment:
Indentation looks off here.

Reply:
I think it is per our standard. It seems the Github diff viewer expands tabs to 4 spaces, which makes it looks incorrect.

Reply:
Turns out you can actually change the tab size setting on GitHub. They set 4 as default but you can change it to 8 at https://github.com/settings/appearance.